### ISSUE TITLE
Components oldData object gets updated and equals data

### ISSUE THREAD

**Issue by IntraNoctem** [aframevr/aframe#5563]
The components oldData object gets updated after the update hook or maybe in next render loop, so that it equals the data object.

That makes it impossible to use oldData in delayed use cases.

- A-Frame Version: 1.6.0
- Platform / Device: any
- Reproducible Code Snippet or URL: https://glitch.com/edit/#!/indecisive-canary-buffalo

**Comment by mrxz** [comment]
This is actually intended behaviour. Prior to 1.6.0 there was a chain of copies `data` -> `oldData` -> `previousOldData` through which updates were propagated. This did mean that in most cases the lifetime of `oldData` passed to the update handler was longer, but no guarantees could be given.

Taking your example, even in 1.5.0 it would report 'incorrect' `oldData` in case >=2 updates to the same component took place either in the task running the `update(oldData)` or in any (micro)task before your timeout handler runs.

Since 1.6.0 this logic was changed to at least guarantee that `oldData` remains valid for the duration of the (synchronous) execution of `update(oldData)`, regardless of any recursive updates. At the same time, instead of leaving the validity of `oldData` up to happenstance, it immediately synchronizes `data` and `oldData` allowing for some additional memory and performance savings.

In short, it's always been unsafe to rely on the `oldData` reference being valid beyond the execution of the `update` handler. Extending the lifetime isn't really feasible as there is no moment where A-Frame can be sure the user is done.

**Comment by IntraNoctem** [comment]
Ok, understood, with this knowledge it makes sense. Thank you very much for your explanation.
